### ISSUE TITLE
FreeBSD 12.0 build fails

### ISSUE THREAD

**Issue by artem00971** [kovidgoyal/kitty#1347]
`make` does not understand Makefile syntax
`gmake` stops with
```python3 setup.py
gmake: python3: Command not found
gmake: *** [Makefile:9: all] Error 127
```
workaround

```ln -s /usr/local/bin/python3.6 ~/bin/python3```

helps, but then

```FileNotFoundError: [Errno 2] No such file or directory:``` 'gcc': 'gcc'
```gmake: *** [Makefile:9: all] Error 1```

I created `CC` environment variable and set it to `clang` then build continues but stops, displaying
```Compiling kittens/choose/main.c ...
Compiling kittens/choose/output.c ...
Compiling kittens/choose/score.c ...
Compiling kittens/choose/unix_compat.c ...
clang -MMD -DNDEBUG -Wextra -Wno-missing-field-initializers -Wall -std=c11 -pedantic-errors -Werror -O3 -fwrapv -fstack-protector-strong -pipe -march=native -fvisibility=hidden -fPIC -flto -pthread -Ikitty -I/usr/local/include/python3.6m -I/usr/local/include/python3.6m -c /usr/home/artem/src/kitty/kittens/choose/output.c -o /usr/home/artem/src/kitty/build/subseq_matcher-output.c.o
In file included from /usr/home/artem/src/kitty/kittens/choose/output.c:8:
In file included from /usr/home/artem/src/kitty/kittens/choose/choose-data-types.h:31:
In file included from /usr/home/artem/src/kitty/kittens/choose/vector.h:9:
In file included from kitty/data-types.h:11:
In file included from /usr/local/include/python3.6m/Python.h:36:
/usr/include/unistd.h:507:19: error: a parameter list without types is only allowed in a
      function definition
char    *fflagstostr(u_long);
                       ^
/usr/include/unistd.h:562:18: error: unknown type name 'fd_set'
int      select(int, fd_set *, fd_set *, fd_set *, struct timeval *);
                     ^
/usr/include/unistd.h:562:28: error: unknown type name 'fd_set'
int      select(int, fd_set *, fd_set *, fd_set *, struct timeval *);
                               ^
/usr/include/unistd.h:562:38: error: unknown type name 'fd_set'
int      select(int, fd_set *, fd_set *, fd_set *, struct timeval *);
                                         ^
/usr/include/unistd.h:580:27: error: unknown type name 'u_long'; did you mean 'long'?
int      strtofflags(char **, u_long *, u_long *);
                              ^
/usr/include/unistd.h:580:37: error: unknown type name 'u_long'; did you mean 'long'?
int      strtofflags(char **, u_long *, u_long *);
                                        ^
In file included from /usr/home/artem/src/kitty/kittens/choose/output.c:8:
In file included from /usr/home/artem/src/kitty/kittens/choose/choose-data-types.h:31:
In file included from /usr/home/artem/src/kitty/kittens/choose/vector.h:9:
In file included from kitty/data-types.h:11:
In file included from /usr/local/include/python3.6m/Python.h:53:
In file included from /usr/local/include/python3.6m/pyport.h:201:
/usr/include/sys/time.h:96:34: error: unknown type name 'u_int'
bintime_mul(struct bintime *_bt, u_int _x)
                                 ^
7 errors generated.
gmake: *** [Makefile:9: all] Error 1
```

**Comment by kovidgoyal** [comment]
Those are all symptoms of ncurses not being able to find the kitty
termfinfo file. kitty uses the TERMINFO env var to point to it. Either
ncurses on BSD does not respect this env var, or it is too old. Install
the terminfo in ~/.terminfo or system wide and you should be fine.

**Comment by artem00971** [comment]
FreeBSD 12 comes with ncurses 5.9, but no `terminfo` database and support utilities installed. By default kitty sets `TERMINFO` variable to `kitty-0.13.3/terminfo` so technically there should be no problem to use kitty, but not in my case.
To solve this I installed port `devel/ncurses` (ncurses-6.1.20190112) which provides single-file hashed database `terminfo.db` and tools , then
```
tic -xs kitty.terminfo
```
to add `xterm-kitty` entry, set `TERMINFO` to `/usr/local/share/misc/terminfo.db`, but it did not help.

I assume this is happening because base system is configured to make software forced to use `termcap` and ignore existence of `terminfo` thus workaround is to build using `devel/ncurses` like some ports do but i have no idea how to build kitty this way.

Also I tried
```
infocmp -Cr xterm-kitty >> /usr/share/misc/termcap && cap_mkdb /usr/share/misc/termcap
```
It solved some problems but still I do not get fully functional terminal.

I guess no kitty for FreeBSD for now.
